src/pages/public/signup/index.js

```diff
@@ -21,33 +15,14 @@ const SignUp = () => {
   };
 
 
-  const submitForm = event => {
-    // console.log(
-    //   arrayOfEmail.find(item => {
-    //     console.log(email + "the email");
-    //     item == email;
-    //   })
-    // );
-    const localData = JSON.parse(localStorage.getItem("user"));
-    let flag = false;
-    localData.map(item => {
-      return (item.Email === email ? flag = true : console.log("flag"))
-    });
+  const submitForm = (event) => {
+    const userDataByEmail = localStorage.getItem('user') ? JSON.parse(localStorage.getItem('user')) : {}
+    userDataByEmail[email] ? (alert('exist'), event.preventDefault()) : userDataByEmail[email] = userDetails
 
+    localStorage.setItem('user', JSON.stringify(userDataByEmail));
```

Comment:
try explaining important or key functions in a component with short comments so that the next devs will find it more easy to go through your code.